### ISSUE TITLE
[UX/ScriptNode]Existing compiled Faust nodes do not detect parameter range changes on subsequent compiles

### ISSUE THREAD

**Issue by aaronventure** [christophhart/HISE#632]
1. Write Faust code, have a few parameters
2. Compile the network to get your compiled Faust node
3. Restart HISE
4. Use the compiled Faust node elsewhere in production networks
5. Change your mind, go back to your Faust code, change the range and stepsize of one or more parameters
6. Compile the node again
7. Restart HISE
8. Existing compiled nodes (e.g. in other networks) do not automatically update the new range/stepsize values for the parameters

**Comment by christoph-hart** [comment]
I wouldn't consider this a bug - once you add a node to a network, the initial parameter ranges are just a default suggestion and are saved as property of the network that loads them. Changing this would severely break backwards compatibility of your projects.

During development you get the bonus function of rebuilding the parameter ranges in Faust, but that's because they are pulled from the Faust code only the first time (so you can add the non-existent properties Skew and StepSize in your network).

**Comment by aaronventure** [comment]
Of course, the parameters get rebuilt manually for the Faust node, but once you compile it, the newly compiled node loaded from the Projects tab will have the correct new range, but existing instances of that same node will not. 

Are you saying that the idea is for the user to manually adjust the new parameter ranges on all the loaded note instances (as the range/stepsize are properties of the network, which does not actually re-instantiate the whole node, just loads the class and sets the ranges that are saved within the network)?

**Comment by christoph-hart** [comment]
> Are you saying that the idea is for the user to manually adjust the new parameter ranges on all the loaded note instances

Yes. How should the network know if the range of a node has been customized by the user or is an old range when it detects a mismatch? The default behaviour should never be to simply overwrite whatever range is found in the network with the current range of the node, otherwise the whole point of storing the custom range in the network is moot.

I could add a function that batch-resets some ranges of some nodes if this is a real world PITA, but then it's a question of UX how to implement this the most useful way.

**Comment by aaronventure** [comment]
You're right, I hadn't considered the fact that they're stored in the network. 

It would be PITA only for a large count of nodes like big matrices, but that's why it's human readable: it can just be changed in the code editor. I'll close this, thanks for addressing it, I genuinely thought it's undesired behavior.
